server/game/cards/13.6-LMHR/FishmongersSquare.js

```diff
@@ -0,0 +1,28 @@
+const DrawCard = require('../../drawcard.js');
+
+class FishmongersSquare extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Draw card and gain gold',
+            phase: 'marshal',
+            cost: ability.costs.kneelSelf(),
+            condition: () => this.opponentDiscardPileHas10(),
+            handler: context => {
+                if(context.player.canDraw()) {
+                    context.player.drawCardsToHand(1);
+                }
+                this.game.addGold(context.player, 1);
+                this.game.addMessage('{0} kneels {1} to draw 1 card and gain 1 gold', context.player, this);

```

Comment:
It might be nice if the message only reflected what happened. i.e. if they only draw, then leave out the gain gold portion of the message. We're not super consistent about this, though, so don't worry about it too much if you find it difficult. My Arya Stark PR (#2736) has an example